### PR TITLE
DOC Add more related projects

### PR DESCRIPTION
**Pull request by rth**
This adds a few more projects to the related projects page: emscripten, iodide, Starboard notebook and basthon.

If you have some wording suggestions don't hesitate to comment. cc @gzuidhof @casatir

### CODE REVIEW

File: docs/project/related-projects.md [hunk 1]
```diff
@@ -1,5 +1,20 @@
 # Related Projects
 
+## WebAssembly ecosystem
+
+- [emscripten](https://emscripten.org/) is a compiler toolchain for WebAssembly
+  that made pyodide possible.
```
Comment (hoodmane) [inline docs/project/related-projects.md:6]:
```suggestion
- [emscripten](https://emscripten.org/) is the compiler toolchain for WebAssembly
  that made Pyodide possible.
```

File: docs/project/related-projects.md [hunk 2]
```diff
@@ -1,5 +1,20 @@
...
+
+## Notebook environements / IDEs / REPLs
+
+ - [Iodide](https://github.com/iodide-project/iodide) is an notebook like
+   environement for literate scientific computing and communication for the
+   web. It is no longer actively developped. Historically, pyodide started
+   as plugin for iodide.
```
Comment (hoodmane) [inline docs/project/related-projects.md:13]:
```suggestion
 - [Iodide](https://github.com/iodide-project/iodide) is a notebook-like
   environment for literate scientific computing and communication for the
   web. It is no longer actively developed. Historically, Pyodide started
   as plugin for iodide.
```

File: docs/project/related-projects.md [hunk 3]
```diff
@@ -1,5 +1,20 @@
...
+ - [Starboard notebook](https://github.com/gzuidhof/starboard-notebook) is an
+   in browser literal notebook runtime that uses pyodide for Python.
```
Comment (hoodmane) [inline docs/project/related-projects.md:15]:
```suggestion
 - [Starboard notebook](https://github.com/gzuidhof/starboard-notebook) is an
   in-browser literal notebook runtime that uses Pyodide for Python.
```

File: docs/project/related-projects.md [hunk 4]
```diff
@@ -1,5 +1,20 @@
...
+ - [Basthon notebook](https://notebook.basthon.fr/) is a static fork of Jupyter
+   notebook with a pyodide kernel (currently in French).
```
Comment (hoodmane) [inline docs/project/related-projects.md:17]:
```suggestion
   notebook with a Pyodide kernel (currently in French).
```

**Review by hoodmane — APPROVED** [review]
Minor spelling / punctuation edits.

**Comment by hoodmane** [comment]
It seems that we don't consistently capitalize "pyodide" in the text in the docs. Maybe I'll open a PR that regularizes capitalization to "Pyodide", and maybe also "Javascript" capitalization.

**Comment by rth** [comment]
Thanks @hoodmane . Looks like my spell checker was off ^^. Yes, I was intentionally not capitalizing pyodide, but it's indeed not consistent in the docs and we should make a decision one way or another.

The general point is that it's easier to write non capitalized names, and I see a lot of libraries that use capitalization in titles but not elsewhere in the documentation e.g. https://setuptools.readthedocs.io/en/latest/userguide/package_discovery.html . For instance the official name of numpy is NumPy, but I don't know any users who write it that way, they docs [is also not consistent](https://numpy.org/doc/stable/reference/internals.html#internal-organization-of-numpy-arrays).

**Comment by hoodmane** [comment]
Okay, I don't have super strong opinion about that, but I think we should try to be consistent. Currently, we capitalize more than not:

pyodide -- 40
Pyodide -- 66

python -- 32
Python -- 147

javascript -- 16
Javascript -- 89
JavaScript -- 13

**Comment by rth** [comment]
yes, let's go with capitalized versions everywhere. I'm sure that some amount of those 40 non capitalized occurrences was me :)

**Pull request merged**